### PR TITLE
chore: convert various files from JS to TS

### PR DESCRIPTION
**Pull request by mhatvan**
#### Description of Change
- update gulpfile to also transpile .tsx files
- convert various files from JS to TS

#### Motivation and Context
Another step to converting the whole codebase from JS to TS

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/api/RecipePreviewsApi.ts [hunk 1]
```diff
@@ -11,7 +13,7 @@ export default class RecipePreviewsApi {
     return this.server.getFeaturedRecipePreviews();
   }
 
-  search(needle) {
+  search(needle: any) {
```
Comment (vraravam) [inline src/api/RecipePreviewsApi.ts:16]: this can be a `string`

File: src/api/RecipesApi.ts [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+  constructor(server: any) {
     this.server = server;
   }
 
   all() {
     return this.server.getInstalledRecipes();
   }
 
-  install(recipeId) {
+  install(recipeId: any) {
```
Comment (vraravam) [inline src/api/RecipesApi.ts:12]: this can be a `string`

File: src/api/ServicesApi.ts [hunk 3]
```diff
@@ -1,30 +1,34 @@
...
     this.server = server;
+    this.local = local;
   }
 
   all() {
     return this.server.getServices();
   }
 
-  create(recipeId, data) {
+  create(recipeId: any, data: any) {
```
Comment (vraravam) [inline src/api/ServicesApi.ts:15]: `recipeId` can be a `string`; `data` - I think we need to create an interface for `Recipe` structure

File: src/api/ServicesApi.ts [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
     return this.server.createService(recipeId, data);
   }
 
-  delete(serviceId) {
+  delete(serviceId: any) {
```
Comment (vraravam) [inline src/api/ServicesApi.ts:19]: this can be a `string`

File: src/api/ServicesApi.ts [hunk 5]
```diff
@@ -1,30 +1,34 @@
...
     return this.server.deleteService(serviceId);
   }
 
-  update(serviceId, data) {
+  update(serviceId: any, data: any) {
```
Comment (vraravam) [inline src/api/ServicesApi.ts:23]: `serviceId` can be a `string`, for `data` we need to create an interface

File: src/api/ServicesApi.ts [hunk 6]
```diff
@@ -1,30 +1,34 @@
...
     return this.server.updateService(serviceId, data);
   }
 
-  reorder(data) {
+  reorder(data: any) {
     return this.server.reorderService(data);
   }
 
-  clearCache(serviceId) {
+  clearCache(serviceId: any) {
```
Comment (vraravam) [inline src/api/ServicesApi.ts:31]: this can be a `string`

File: src/api/UserApi.ts [hunk 7]
```diff
@@ -1,39 +1,44 @@
...
+
+  local: any;
+
+  constructor(server: any, local: any) {
     this.server = server;
     this.local = local;
   }
 
-  login(email, password) {
+  login(email: any, password: BinaryLike) {
```
Comment (vraravam) [inline src/api/UserApi.ts:14]: `email` can be a `string` or is there a specialized type for Email?

File: src/api/UserApi.ts [hunk 8]
```diff
@@ -1,39 +1,44 @@
...
-  signup(data) {
+  signup(data: { password: BinaryLike }) {
     Object.assign(data, {
       password: hash(data.password),
     });
     return this.server.signup(data);
   }
 
-  password(email) {
+  password(email: any) {
```
Comment (vraravam) [inline src/api/UserApi.ts:29]: this can be a `string` or is there a specialized type for Email?

**Review by vraravam — CHANGES_REQUESTED** [review]

File: src/api/UserApi.ts (hunk 7 above)
Comment (mhatvan) [reply src/api/UserApi.ts:14]: `string` is correct, I updated it in another commit. 

Note that I auto-inferred all params after converting the file to `.ts`, so adding the correct types was of no importance to me here, can be done in a final step after the conversion to add proper types for all `any`.

**Review by vraravam — APPROVED** [review]

**Pull request merged**